### ISSUE TITLE
[v2.22.0] Typing issue: Cannot redeclare block-scoped variable daysInWeek/minTime

### ISSUE THREAD

**Issue by Rogdham** [date-fns/date-fns#2491]
Hello @kossnocorp, in the v2.22.0 release, there are duplicates of `daysInWeek` and `minTime` in `typings.d.ts` that create issues with TypeScript.

<details><summary>Unfold for error messages</summary>
<p>

```
node_modules/date-fns/typings.d.ts:1131:9 - error TS2451: Cannot redeclare block-scoped variable 'daysInWeek'.

1131   const daysInWeek: number
             ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:1133:9 - error TS2451: Cannot redeclare block-scoped variable 'daysInWeek'.

1133   const daysInWeek: number
             ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:1141:9 - error TS2451: Cannot redeclare block-scoped variable 'minTime'.

1141   const minTime: number
             ~~~~~~~

node_modules/date-fns/typings.d.ts:1143:9 - error TS2451: Cannot redeclare block-scoped variable 'minTime'.

1143   const minTime: number
             ~~~~~~~

node_modules/date-fns/typings.d.ts:5598:9 - error TS2451: Cannot redeclare block-scoped variable 'daysInWeek'.

5598   const daysInWeek: number
             ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:5600:9 - error TS2451: Cannot redeclare block-scoped variable 'daysInWeek'.

5600   const daysInWeek: number
             ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:5608:9 - error TS2451: Cannot redeclare block-scoped variable 'minTime'.

5608   const minTime: number
             ~~~~~~~

node_modules/date-fns/typings.d.ts:5610:9 - error TS2451: Cannot redeclare block-scoped variable 'minTime'.

5610   const minTime: number
             ~~~~~~~

node_modules/date-fns/typings.d.ts:10231:9 - error TS2451: Cannot redeclare block-scoped variable 'daysInWeek'.

10231   const daysInWeek: number
              ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:10233:9 - error TS2451: Cannot redeclare block-scoped variable 'daysInWeek'.

10233   const daysInWeek: number
              ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:10241:9 - error TS2451: Cannot redeclare block-scoped variable 'minTime'.

10241   const minTime: number
              ~~~~~~~

node_modules/date-fns/typings.d.ts:10243:9 - error TS2451: Cannot redeclare block-scoped variable 'minTime'.

10243   const minTime: number
              ~~~~~~~

node_modules/date-fns/typings.d.ts:14698:9 - error TS2451: Cannot redeclare block-scoped variable 'daysInWeek'.

14698   const daysInWeek: number
              ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:14700:9 - error TS2451: Cannot redeclare block-scoped variable 'daysInWeek'.

14700   const daysInWeek: number
              ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:14708:9 - error TS2451: Cannot redeclare block-scoped variable 'minTime'.

14708   const minTime: number
              ~~~~~~~

node_modules/date-fns/typings.d.ts:14710:9 - error TS2451: Cannot redeclare block-scoped variable 'minTime'.

14710   const minTime: number
              ~~~~~~~

node_modules/date-fns/typings.d.ts:22013:3 - error TS2300: Duplicate identifier 'daysInWeek'.

22013   daysInWeek: number
        ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:22015:3 - error TS2300: Duplicate identifier 'daysInWeek'.

22015   daysInWeek: number
        ~~~~~~~~~~

node_modules/date-fns/typings.d.ts:22023:3 - error TS2300: Duplicate identifier 'minTime'.

22023   minTime: number
        ~~~~~~~

node_modules/date-fns/typings.d.ts:22025:3 - error TS2300: Duplicate identifier 'minTime'.

22025   minTime: number
        ~~~~~~~
```

</p>
</details>

---

### FAQ

This issue is getting some traction, people are subscribing to it, so we would all appreciate if you could reduce the noise and comment only if you bring something new to the discussion :pray: 

#### I'm having the same issue

 - please react with the :+1: emoji on this message instead of posting a message to say that
 - also the errors and impacted files are listed above (if you unfold when mentionned)

#### How to fix the issue for now?

 - you can revert to previous version 2.21.3 (or maybe earlier, [some user reported](https://github.com/date-fns/date-fns/issues/2491#issuecomment-850282661) the issue was present in 2.21.3)
 - you can use the following workaround: set [`skipLibCheck`](https://www.typescriptlang.org/tsconfig#skipLibCheck) to `true` in `tsconfig.json` (section `compilerOptions`).

#### I want to fix the issue by creating a PR

:hourglass_flowing_sand: [People are looking to fix that issue now](https://github.com/date-fns/date-fns/issues/2491#issuecomment-850273436) :crossed_fingers: 

~Please coordinate by writing below! Note that the `typings.d.ts` file is generated automatically by `scripts/build/typings.js`, so you probably need to look into that and make the changes there.~

**Comment by Marcuss17** [comment]
Also having this issue

**Comment by swftvsn** [comment]
Just got this too.

**Comment by Gyufei** [comment]
![image](https://user-images.githubusercontent.com/25930828/119952286-6e9f4800-bfcf-11eb-9256-10b169721120.png)

**Comment by abhilash-vadlamudi** [comment]
also having the issue

**Comment by douweknook** [comment]
I'm having the same problem. The issue seems to be that several constants are defined twice in the `typings.d.ts` file. It's all these lines:
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L1131-L1133
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L1143-L1145
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L5598-L5600
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L5608-L5610
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L10231-L10233
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L10241-L10243
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L14698-L14700
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L14708-L14710
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L22013-L22015
https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L22023-L22025
Always related to `daysInWeek` and `minTime`.

I could try to make a PR to fix it but I'm not sure if the types are generated from the code and then why the duplication occurs.

**Comment by lpellegr** [comment]
@kossnocorp kindly ping as many people updating date-fns will experience errors.

**Comment by Rogdham** [comment]
@douweknook The typings.d.ts file is generated automatically by `scripts/build/typings.js`, so you probably need to look into that and make the changes there.

_I've edited the first message to mention that_

**Comment by tan75** [comment]
hi everyone,
we are looking into this issue now.

**Comment by vibhakara-acharya** [comment]
We are facing same issue due to which our prod pipeline is failing. As i looked into it, this caused by recent commit 3hrs ago typings.d.ts

![image](https://user-images.githubusercontent.com/42801902/119960637-ca170900-bfc2-11eb-9e2e-d330dc42864d.png)

Please look into it.

**Comment by AniaMakes** [comment]
Just to add that we're seeing this issue when using v2.21.3

**Comment by Rogdham** [comment]
> Just to add that we're seeing this issue when using v2.21.3

@AniaMakes good to know, although it seems fine on my side with v2.21.3; I'm updating my first message regardless

**Comment by blazmrak** [comment]
I've deployed my app around 1h before 2.22.0 and deployment went through, then after 2.22.0 was released the build failed. When I lowered the version back to 2.21.3 the build went through normally.

**Comment by kossnocorp** [comment]
@Rogdham, thank you so much for a promptly and carefully crafted issue. I'm already working on a fix and hope to release it really soon.

**Comment by fturmel** [comment]
@kossnocorp see #2494, I think that should take care of the problem. That constants file needs explicit @name jsdocs, Babel is mangling the output and jsdocs can't keep track of what variable is associated with which jsdoc block.

**Comment by kossnocorp** [comment]
Fixed and released in `v2.22.1`
